tests/Zend/Db/TestSetup.php

```diff
@@ -80,14 +80,18 @@ protected function _setUpTestUtil()
      */
     protected function _setUpAdapter()
     {
-        $this->_db = Zend_Db::factory($this->getDriver(), $this->_util->getParams());
         try {
+            $this->_db = Zend_Db::factory($this->getDriver(), $this->_util->getParams());
             $conn = $this->_db->getConnection();
         } catch (Zend_Exception $e) {
             $this->_db = null;
             $this->assertTrue($e instanceof Zend_Db_Adapter_Exception,
                 'Expecting Zend_Db_Adapter_Exception, got ' . get_class($e));
             $this->markTestSkipped($e->getMessage());
+        } catch (Throwable $e) {
+            if (stristr($e->getMessage(), 'undefined constant')) {
+                $this->markTestSkipped();
+            }
```

Comment:
In which case this the db factory throws this `undefined constant` error? What kind of constants were missing?

Reply:
DB2 constants